From 6c8eb4100cfe557f9f19c04f2fc862c0e1779f42 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:03 -0400
Subject: [usb] cdc-acm: fix shutdown and suspend race

Message-id: <1410887463-20674-472-git-send-email-dzickus@redhat.com>
Patchwork-id: 95505
O-Subject: [RHEL7 PATCH 471/591] USB: cdc-acm: fix shutdown and suspend race
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit ed797074031a37bb9bf4a70952fffc606b77274d
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:40 2014 +0200

    USB: cdc-acm: fix shutdown and suspend race

    We should stop I/O unconditionally at suspend rather than rely on the
    tty-port initialised flag (which is set prior to stopping I/O during
    shutdown) in order to prevent suspend returning with URBs still active.

    Fixes: 11ea859d64b6 ("USB: additional power savings for cdc-acm devices
    that support remote wakeup")

    Cc: <stable@vger.kernel.org> # v2.6.27
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-acm.c b/drivers/usb/class/cdc-acm.c
index 2258827..1ac6c5d 100644
--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -1550,8 +1550,7 @@ static int acm_suspend(struct usb_interface *intf, pm_message_t message)
  if (cnt)
   return 0;
 
- if (test_bit(ASYNCB_INITIALIZED, &acm->port.flags))
-  stop_data_traffic(acm);
+ stop_data_traffic(acm);
 
  return 0;
 }
-- 
1.7.1